StitchCoreServicesMongoDbRemote-iOS/StitchCoreServicesMongoDbRemote-iOS/RemoteMongoClient.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import StitchCore
+import StitchCore_iOS
+import StitchCoreServicesMongoDbRemote
+
+private final class RemoteMongoDbServiceClientFactory: NamedServiceClientFactory {
+    typealias ClientType = RemoteMongoClient
+    
+    func client(withServiceClient serviceClient: StitchServiceClient,
+                withClientInfo clientInfo: StitchAppClientInfo) -> RemoteMongoClient {
+        return RemoteMongoClient.init(
+            withClient: CoreRemoteMongoClient.init(withService: serviceClient),
+            withDispatcher: OperationDispatcher.init(withDispatchQueue: DispatchQueue.global())
+        )
+    }
+}
+
+/**
+ * A class which can be used to get database and collection objects which can be used to interact with MongoDB data via
+ * the Stitch MongoDB service.
+ */
+public class RemoteMongoClient {
+    private let dispatcher: OperationDispatcher
+    private let proxy: CoreRemoteMongoClient
+    
+    internal init(withClient client: CoreRemoteMongoClient,
+                  withDispatcher dispatcher: OperationDispatcher) {
+        self.proxy = client
+        self.dispatcher = dispatcher
+    }
+    
+    /**
+     * Gets a `CoreRemoteMongoDatabase` instance for the given database name.
+     *
+     * - parameter name: the name of the database to retrieve
+     */
+    public func db(_ name: String) -> RemoteMongoDatabase {
```

Comment:
We should call this database.

Reply:
This is following suit with the Swift driver. If we decide to deviate with respect to arg labels, I'll change this too.

Reply:
Keeping it as `db` per the discussion we just had in the SDK chat